cpp/include/cuml/neighbors/knn_api.h

```diff
@@ -16,11 +16,8 @@
 
 #pragma once
 
-#include <cuml_api.h>
-
-#ifdef __cplusplus
```

Comment:
same as dbscan_api.h. Since this is going to be a C-header file, I believe we should keep the `__cplusplus` macro guard to avoid symbol-mangling.

Reply:
I took this into consideration and updated the design to revert these changes. See this comment for more info: https://github.com/rapidsai/cuml/pull/3402#issuecomment-772092988